lib/rubocop/cop/lint/format_parameter_mismatch.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # This lint sees if there is a mismatch between the number of
+      # expected parameters for format and what is actually passed.
+      class FormatParameterMismatch < Cop
+
+        VALID_FIELDS_REGEX = /%[bBdiouxXeEfgGacps]/
+
+        def on_send(node)
+          add_offense(node, :selector) if offending_node?(node)
+        end
+
+        private
+
+        def offending_node?(node)
+          (sprintf?(node) || format?(node)) && missing_parameters?(node)
+        end
+
+        def format_method?(name, node)
+          receiver, method_name, *args = *node
+
+          # commands have no explicit receiver
+          unless !receiver && method_name == name
+            false
+          else
+            args.size > 1
+          end
+        end
+
+        def expected_fields(args)
+          args
+            .first
+            .loc
+            .expression
+            .source
+            .scan(VALID_FIELDS_REGEX)
+        end
+
+        def missing_parameters?(node)
```

Comment:
Technically speaking, those would be missing arguments. We talk about params in a definition and about args in an invocation.